FiiPrezent/UnitOfWork.cs

```diff
@@ -0,0 +1,35 @@
+using System.Threading.Tasks;
+using FiiPrezent.Data;
+using FiiPrezent.Entities;
+using FiiPrezent.Interfaces;
+
+namespace FiiPrezent
+{
+    public class UnitOfWork : IUnitOfWork
```

Comment:
Keep it simple.. db context-ul din EF face exact acelasi lucru.

Reply:
Eu am incercat sa abstractizez EF de aplicatie. De asta am folosit si repositoryul

Reply:
M-am uitat mai atent. Implementarea de UoW e foarte buna. Ce as face diferit e ca nu as folosi unit of work direct din controllere. Sunt mai multe variante aici. Commands and Queries ar fi una  (see https://martinfowler.com/bliki/CQRS.html)